internal/bminventory/inventory.go

```diff
@@ -2627,7 +2624,7 @@ func validateUserManagedNetworkConflicts(params *models.ClusterUpdateParams, sin
 		log.WithError(err)
 		return common.NewApiError(http.StatusBadRequest, err)
 	}
-	if params.MachineNetworks != nil && len(params.MachineNetworks) > 0 && !singleNodeCluster {
+	if common.IsNetworksSliceNonEmpty(params.MachineNetworks) && !singleNodeCluster {
```

Comment:
This `if` got simplified because `IsNetworksSliceNonEmpty` handles the case of getting `nil` as an argument and simply returns `False`